### DISCUSSION TITLE
Custom frontend design for miningcore - Allow user to set their minimum payout?

### DISCUSSION THREAD

**Discussion by blackmennewstyle** [coinfoundry/miningcore#1482] [Q&A] [answered]
Hello guys,

First of all, i have to say i'm very impressed by `miningcore`. It's an incredible project.

I would like to design my own custom frontend for it and i have a question: Does `miningcore` in its current stage of developpment allow implementation for features like `custom mininmum payout` per user/miner/wallet?

**Comment by oliverw** [comment, marked as answer]
Yes thats supported through the `{poolId}/miners/{address}/settings` API route.

**Reply by blackmennewstyle** [reply]
Thanks so much for your answer :fire: :rocket:

**Reply by oliverw** [reply]
> Thanks so much for your answer 🔥 🚀

You're welcome.

**Reply by hjonathan2019** [reply]
@blackmennewstyle I would like to do something similar, could you help me to achieve it?

**Comment by blackmennewstyle** [comment]
Hello,

Sorry to bother you again but i can not seem to understand how the `{poolId}/miners/{address}/settings` API route is actually working.

Am i correct to assume that the following portion of code is responsible for settings update?
```
[HttpPost("{poolId}/miners/{address}/settings")]
    public async Task<Responses.MinerSettings> SetMinerSettingsAsync(string poolId, string address,
        [FromBody] Requests.UpdateMinerSettingsRequest request, CancellationToken ct)
    {
        var pool = GetPool(poolId);

        if(string.IsNullOrEmpty(address))
            throw new ApiException("Invalid or missing miner address", HttpStatusCode.NotFound);

        if(pool.Template.Family == CoinFamily.Ethereum)
            address = address.ToLower();

        if(request?.Settings == null)
            throw new ApiException("Invalid or missing settings", HttpStatusCode.BadRequest);

        if(!IPAddress.TryParse(request.IpAddress, out var requestIp))
            throw new ApiException("Invalid IP address", HttpStatusCode.BadRequest);

        // fetch recent IPs
        var ips = await cf.Run(con=> shareRepo.GetRecentyUsedIpAddressesAsync(con, null, poolId, address, ct));

        // any known ips?
        if(ips == null || ips.Length == 0)
            throw new ApiException("Address not recently used for mining", HttpStatusCode.NotFound);

        // match?
        if(!ips.Any(x=> IPAddress.TryParse(x, out var ipAddress) && ipAddress.IsEqual(requestIp)))
            throw new ApiException("None of the recently used IP addresses matches the request", HttpStatusCode.Forbidden);

        // map settings
        var mapped = mapper.Map<Persistence.Model.MinerSettings>(request.Settings);

        // clamp limit
        if(pool.PaymentProcessing != null)
            mapped.PaymentThreshold = Math.Max(mapped.PaymentThreshold, pool.PaymentProcessing.MinimumPayment);

        mapped.PoolId = pool.Id;
        mapped.Address = address;

        // finally update the settings
        return await cf.RunTx(async (con, tx) =>
        {
            await minerRepo.UpdateSettingsAsync(con, tx, mapped);

            logger.Info(()=> $"Updated settings for pool {pool.Id}, miner {address}");

            var result = await minerRepo.GetSettingsAsync(con, tx, mapped.PoolId, mapped.Address);
            return mapper.Map<Responses.MinerSettings>(result);
        });
    }
```
So i should send two variables with request 'POST': `Settings` and `IPAddress`?

I understand the later one, but what is the first one?

**Reply by blackmennewstyle** [reply]
Nevermind,

I found the answer here: https://miningcore.pro/swagger/index.html#/pools/PoolApi_MinerSettings

**Comment by blackmennewstyle** [comment]
@oliverw Thanks to your help, i managed to create my own `frontend` for `miningcore`.

https://ergo.cedric-crispin.com/
https://monero.cedric-crispin.com/
https://vertcoin.cedric-crispin.com/

I will definitely not win some web-design prizes anytime soon, but it was a very fun programming endeavor for me lol

Have a great week and keep it up the great job! Looking forward to the next features and hoping to be able to contribute even a little bit :)

**Reply by oliverw** [reply]
Great and thanks for the kind words. Good luck.
